pkgs/development/tools/picotool/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, stdenv, fetchFromGitHub, cmake, pkg-config, libusb }:
+
+stdenv.mkDerivation rec {
+  pname = "picotool";
+  version = "1.0.1";
+
+  src = fetchFromGitHub {
+    owner = "raspberrypi";
+    repo = pname;
+    rev = version;
+    sha256 = "1k5j742sj91akdrgnd3wa5csqb638dgaz0c09zsr22fcqz0qhzig";
+  };
+
+  pico-sdk = fetchFromGitHub {
```

Comment:
This should probably be an extra package.

Reply:
I tried to package pico-sdk on openSUSE before and it doesn't beneficial at all. Also it only required by picotool. For development, adding pico-sdk as submodule will be ok. Most of the people will use arduino-core port anyways.

Reply:
> Also it only required by picotool.

I am pretty sure NixOS has lots of packages like this.

> For development, adding pico-sdk as submodule will be ok.

Using a submodule is actually worse because it makes updating it harder.

Adding it as an extra package makes it easier to update it, reuse it anytime later and patching it.

Reply:
Created #125648

Reply:
You can cherry-pick the commit into this PR.